lib/dynamoid/adapter.rb

```diff
@@ -155,9 +155,15 @@ def delete_table(table_name, options = {})
     #
     # @since 0.2.0
     def method_missing(method, *args, &block)
-      return benchmark(method, *args) { adapter.send(method, *args, &block) } if adapter.respond_to?(method)
+      super unless adapter.respond_to?(method)
 
-      super
+      if args.last.is_a?(Hash)
+        return benchmark(method, *args[0..-2], **args.last) do
+          adapter.send(method, *args[0..-2], **args.last, &block)
+        end
+      end
+
+      benchmark(method, *args) { adapter.send(method, *args, &block) }
```

Comment:
Again, what do you think about changing method signature?

```
def method_missing(method, *args, **kwargs, &block)
```

Reply:
Thank you for the review and feedback. Looks as is using `ruby2_keywords` is the way to go with [defining a dummy version of it for Ruby 2.6 and prior](https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/) (thanks @eregon):

```ruby
def ruby2_keywords(*)
end if RUBY_VERSION < "2.7"
```

Reply:
Yeah. Looks like it's the easiest way.

But comparing string versions `RUBY_VERSION < "2.7"` isn't correct for minor versions like '2.10' and greater. I like the solution from https://github.com/ruby/ruby2_keywords:

```ruby
class Module
  unless respond_to?(:ruby2_keywords, true)
    private
    def ruby2_keywords(name, *)
      # nil
    end
  end
end
```

But I would like to avoid monkey matching global classes.